CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -76,26 +71,28 @@ class CardClient internal constructor(
         approveOrderId = cardRequest.orderId
         analytics.notifyApproveOrderStarted(cardRequest.orderId)
 
-        CoroutineScope(dispatcher).launch(approveOrderExceptionHandler) {
-            // TODO: migrate away from throwing exceptions to result objects
-            try {
-                val response = checkoutOrdersAPI.confirmPaymentSource(cardRequest)
-                if (response.payerActionHref == null) {
-                    analytics.notifyApproveOrderSucceeded(response.orderId)
-                    val result =
-                        response.run { CardResult(orderId = orderId, status = status?.name) }
-                    approveOrderListener?.onApproveOrderSuccess(result)
-                } else {
-                    analytics.notifyApproveOrderAuthChallengeReceived(cardRequest.orderId)
-                    approveOrderListener?.onApproveOrderThreeDSecureWillLaunch()
+        CoroutineScope(dispatcher).launch {
+            when (val response = checkoutOrdersAPI.confirmPaymentSource(cardRequest)) {
+                is ConfirmPaymentSourceResult.Success -> {
+                    if (response.payerActionHref == null) {
+                        analytics.notifyApproveOrderSucceeded(response.orderId)
+                        val result =
+                            response.run { CardResult(orderId = orderId, status = status?.name) }
+                        approveOrderListener?.onApproveOrderSuccess(result)
+                    } else {
+                        analytics.notifyApproveOrderAuthChallengeReceived(cardRequest.orderId)
```

Comment:
Do you check formatting of the returnURL? In iOS, we have this code that checks formatting https://github.com/paypal/paypal-ios/blob/60f2f95eab44048211b5289e8fb5b87538daa751/Sources/CardPayments/CardClient.swift#L108

Reply:
We don't actually on Android. Is this something we can omit?